src/core/geometry/qgstriangle.cpp

```diff
@@ -224,6 +223,32 @@ bool QgsTriangle::fromWkt( const QString &wkt )
   return true;
 }
 
+QDomElement QgsTriangle::asGML2( QDomDocument &doc, int precision, const QString &ns ) const
+{
+  if ( !isEmpty() )  /* else crash */
```

Comment:
Better move this check up to QgsCurvePolygon::asGML2 - it will be because qgscurvepolygon.cpp:336 is dereferencing an empty exterior ring (and I think the right approach is to skip the outerBoundaryIs element when the exterior ring is missing).